md2man/roff_test.go

```diff
@@ -305,8 +333,12 @@ func execRecoverableTestSuite(t *testing.T, tests []string, params TestParams, s
 
 func runMarkdown(input string, params TestParams) string {
 	renderer := NewRoffRenderer()
+	extensions := params.extensions
+	if extensions == 0 {
+		extensions = renderer.extensions
+	}
```

Comment:
So, after digging why it didn't work with the `% xxx` pandoc headers, this seemed to be the problem; the tests were running without the `blackfriday.Titleblock` extension enabled, so blackfriday was not converting them to H1 blocks.